docs/reference/changelog.rst

```diff
@@ -4,6 +4,16 @@
 Changelog
 *********
 
+5.8.0 (upcoming)
+----------------
+
+Application
+===========
+
+- Add a new ``for`` statement for advanced grammar to select against a platform.
```

Comment:
```suggestion
- Add the ``for`` selector to the YAML grammar. With it, crafters can set different 
  values depending on the active platform.
```

I have mixed feelings about "select against".

We can drop "YAML" if it feels redundant.